python/pyspark/sql/functions.py

```diff
@@ -3925,10 +3987,20 @@ def dayofmonth(col: "ColumnOrName") -> Column:
 
 def dayofyear(col: "ColumnOrName") -> Column:
     """
-    Extract the day of the year of a given date as integer.
+    Extract the day of the year of a given date/timestamp as integer.
 
     .. versionadded:: 1.5.0
 
+    Parameters
+    ----------
+    col : :class:`~pyspark.sql.Column` or str
+        target date/timestamp column to work on.
+
+    Returns
+    -------
+    :class:`~pyspark.sql.Column`
+        day of the year for given date/timestamp as integer.
```

Comment:
Very nit, and don't much have strong feeling about this,

Maybe:

```
`day` of the year for given...
```

since we describe

```
 `week` of the year for given...
```

in the `weekofyear` function.

Reply:
But let's keep it as is for now.

Thanks for working on this!

Reply:
feel free to create a followup @itholic

Reply:
would be great to fix other typos or nits together if you find some more

Reply:
Sure, let me find some more improvements and will open a followup.